code/modules/transport/tram/tram_floors.dm

```diff
@@ -0,0 +1,243 @@
+			to_chat(user, span_warning("The security screws need to be removed first!"))
+
+	else
+		if(tool.tool_behaviour == TOOL_SCREWDRIVER)
+			user.visible_message(span_notice("[user] begins to fasten the tile..."),
+									span_notice("You begin to fasten the tile..."))
+			if(tool.use_tool(src, user, 4 SECONDS, volume = 50))
+				secured = TRUE
+				to_chat(user, span_notice("The tile is securely screwed in place."))
+
+		else if(tool.tool_behaviour == TOOL_CROWBAR)
+			user.visible_message(span_notice("[user] wedges \the [tool] into the tile's gap in the edge and starts prying..."),
+									span_notice("You wedge \the [tool] into the tram panel's gap in the frame and start prying..."))
+			if(tool.use_tool(src, user, 4 SECONDS, volume = 50))
+				to_chat(user, span_notice("The panel pops out of the frame."))
+				var/obj/item/stack/thermoplastic/pulled_tile = new()
+				pulled_tile.update_integrity(atom_integrity)
+				user.put_in_hands(pulled_tile)
+				qdel(src)
+
+	if (tool.tool_behaviour)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	return ..()
+
+/obj/structure/thermoplastic/welder_act(mob/living/user, obj/item/tool)
+	if(atom_integrity >= max_integrity)
+		to_chat(user, span_warning("[src] is already in good condition!"))
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	if(!tool.tool_start_check(user, amount = 0))
+		return FALSE
+	to_chat(user, span_notice("You begin repairing [src]..."))
+	var/integrity_to_repair = max_integrity - atom_integrity
+	if(tool.use_tool(src, user, integrity_to_repair, volume = 50))
+		atom_integrity = max_integrity
+		to_chat(user, span_notice("You repair [src]."))
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
+/obj/item/stack/thermoplastic
+	name = "thermoplastic tram tile"
+	singular_name = "thermoplastic tram tile"
+	desc = "A high-traction floor tile. It sparkles in the light."
+	icon = 'icons/obj/tiles.dmi'
+	lefthand_file = 'icons/mob/inhands/items/tiles_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/tiles_righthand.dmi'
+	icon_state = "tile_textured_white_large"
+	inhand_icon_state = "tile-tile_textured_white_large"
+	color = COLOR_TRAM_BLUE
+	w_class = WEIGHT_CLASS_NORMAL
+	force = 1
+	throwforce = 1
+	throw_speed = 3
+	throw_range = 7
+	max_amount = 60
+	novariants = TRUE
+	merge_type = /obj/item/stack/thermoplastic
+	var/tile_type = /obj/structure/thermoplastic
+	/// Cached associative lazy list to hold the radial options for tile reskinning. See tile_reskinning.dm for more information. Pattern: list[type] -> image
+	var/list/tile_reskin_types = list(
+		/obj/item/stack/thermoplastic/light,
+	)
+
+/obj/item/stack/thermoplastic/light
+	color = COLOR_TRAM_LIGHT_BLUE
+	tile_type = /obj/structure/thermoplastic/light
+
+/obj/item/stack/thermoplastic/Initialize(mapload, new_amount, merge = TRUE, list/mat_override=null, mat_amt=1)
+	. = ..()
+	pixel_x = rand(-3, 3)
+	pixel_y = rand(-3, 3) //randomize a little
+	//if(tile_reskin_types)
+	//	tile_reskin_types = tile_reskin_list(tile_reskin_types)
```

Comment:
?

Reply:
Some leftover WIP where I was thinking of adding re-skinnable tram tiles.